hammr/commands/migration/migration.py

```diff
@@ -48,8 +60,170 @@ def do_list(self, args):
+        return doParser
+
+    def do_launch(self, args):
+        try:
+            # add arguments
+            doParser = self.arg_launch()
+
+            try:
+                doArgs = doParser.parse_args(shlex.split(args))
+            except SystemExit:
+                return 2
+
+            migrationConfig = migration_utils.check_migration_configuration(doArgs.file)
+            targetFormat, targetPlatform = migration_utils.check_target(self.api, self.login, migrationConfig)
+            myPublishImage = migration_utils.check_publish_image(migrationConfig, targetFormat, doArgs.file)
+            myCredAccount = migration_utils.check_credAccount(self.api, self.login, migrationConfig, targetPlatform)
+
+            printer.out("file: " + doArgs.file + " is validated", "OK")
+
+            myMigration = self.createMigration(migrationConfig, targetFormat, myCredAccount, myPublishImage)
+
+            # call UForge API
+            self.api.Users(self.login).Migrations.Create(body=myMigration, element_name="ns1:migration")
+
+            # download migration binary
+            local_uforge_migration_path = self.download_migration_binary()
+
+            r_code = self.deploy_and_launch_agent(self.login, self.password, migrationConfig,
+                                                  local_uforge_migration_path, self.api.getUrl())
+
+            if r_code != 0:
+                return
+            printer.out(
+                "Migration launched successfully, please go to the platform to follow steps of the migration.",
+                printer.OK)
+
+        except ArgumentParserError as e:
+            printer.out("ERROR: In Arguments: " + str(e), printer.ERROR)
+            self.help_launch()
+
+        except Exception as e:
+            if hammr_utils.is_uforge_exception(e):
+                return hammr_utils.handle_uforge_exception(e)
+            printer.out(str(e), printer.ERROR)
+            # return hammr_utils.handle_uforge_exception(e)
+
+    def help_launch(self):
+        doParser = self.arg_launch()
+        doParser.print_help()
+
+    def download_migration_binary(self):
+        uri = generics_utils.get_uforge_url_from_ws_url(self.api.getUrl())
+        download_url = uri + constants.URI_MIGRATION_BINARY
+        if os.path.isdir(constants.TMP_WORKING_DIR):
+            # delete tmp dir
+            shutil.rmtree(constants.TMP_WORKING_DIR)
+        os.mkdir(constants.TMP_WORKING_DIR)
+        local_uforge_migration_path = constants.TMP_WORKING_DIR + os.sep + constants.MIGRATION_BINARY_NAME
+
+        dlUtils = download_utils.Download(download_url, local_uforge_migration_path,
+                                          not self.api.getDisableSslCertificateValidation())
+        try:
+            dlUtils.start()
+        except Exception:
+            return
+        return local_uforge_migration_path
+
+    def deploy_and_launch_agent(self, uforge_login, uforge_password, migrationConfig, file_src_path, uforge_url):
```

Comment:
this method contains duplicated code from scan.py, it should be extracted and reuse